### ISSUE TITLE
[Bug] type='build' specs conflict when they shouldn't

### ISSUE THREAD

**Issue by citibeth** [spack/spack#1768]
I have the following situation:
1. `py-scipy` seems to require `binutils@2.26` to build.

```
        depends_on('binutils@2.26:', type='build')
```
1. `py-pillow` has problems with @2.26, it seems to required `binutils@:2.25` to build.

```
    depends_on('binutils@:2.25', type='build')   # Doesn't work with 2.26
```

This should cause no problem because these are build-only dependencies.  However, Spack barfs on it:

```
$ spack spec py-giss
Input spec
------------------------------
  py-giss

Normalized
------------------------------
==> Error: Invalid spec: 'binutils@2.26:^bison^flex^m4'. Package binutils requires version :2.25, but spec asked for 2.26:
```

**Comment by davydden** [comment]
see https://github.com/LLNL/spack/issues/839

**Comment by adamjstewart** [comment]
This should be fixed by #2548.

**Comment by alalazo** [comment]
Duplicate of #839
